src/utils/liquidityPools.js

```diff
@@ -126,10 +144,25 @@ export const getAddLiquidityTransactions = async (
   let addLiquidityTransactions = [];
 
   const addApproveTransaction = async (tokenAmount, tokenAmountBN, tokenAsset) => {
-    const erc20Contract = getContract(tokenAsset.address, ERC20_CONTRACT_ABI);
-    const approvedAmountBN = erc20Contract
-      ? await erc20Contract.allowance(sender, UNISWAP_ROUTER_ADDRESS)
-      : null;
+    const erc20Contract = etherspotService.getContract<?EtherspotErc20Interface>(
+      CHAIN.ETHEREUM,
+      ERC20_CONTRACT_ABI,
+      tokenAsset.address,
+    );
+
+    if (!erc20Contract) {
+      reportErrorLog('getAddLiquidityTransactions failed: no erc20Contract on addApproveTransaction', {
+        tokenAsset,
+      });
+      return;
+    }
+
+    let approvedAmountBN;
+    try {
+      approvedAmountBN = await erc20Contract.callAllowance(sender, UNISWAP_ROUTER_ADDRESS);
+    } catch (error) {
+      reportErrorLog('getAddLiquidityTransactions failed: callAllowance failed', { tokenAsset, error });
+    }
 
     if (!approvedAmountBN || tokenAmountBN.gt(approvedAmountBN)) {
```

Comment:
Seems like return type is now ethers big number. Does it still work correctly?

Reply:
It was ethers BigNumber before as well since it was using ethers Contract interface directly and now the return is wrapper through Etherspot SDK.